src/librustc_mir/borrow_check/error_reporting.rs

```diff
@@ -198,19 +198,43 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
                 let place = &self.move_data.move_paths[mpi].place;
 
                 let ty = place.ty(self.mir, self.infcx.tcx).to_ty(self.infcx.tcx);
-                let note_msg = match self.describe_place_with_options(
-                    place,
-                    IncludingDowncast(true),
-                ) {
-                    Some(name) => format!("`{}`", name),
+                let opt_name = self.describe_place_with_options(place, IncludingDowncast(true));
+                let note_msg = match opt_name {
+                    Some(ref name) => format!("`{}`", name),
                     None => "value".to_owned(),
                 };
-
-                err.note(&format!(
-                    "move occurs because {} has type `{}`, \
-                     which does not implement the `Copy` trait",
-                    note_msg, ty
-                ));
+                let mut note = true;
+                for decl in &self.mir.local_decls {
```

Comment:
This is not very efficient, but 1) will only happen in case there's an error being emitted (no performance impact in happy path) 2) should be a short `n` to get to the appropriate `decl` and 3) it could be safely be removed while keeping the code at line 220.

If we wanted to modify this, we could introduce a `HashMap` for this, but would be a memory use increase in _all_ cases.